sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/FilteredAttributes.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.view;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.internal.ImmutableKeyValuePairs;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+
+@SuppressWarnings("unchecked")
+final class FilteredAttributes implements Attributes {
+
+  private final Object[] sourceData;
+  private final BitSet filteredIndices;
```

Comment:
Bitset is backed by long[]. The smallest possible long array still allocated > 8 bytes (64 bits), which is many more than needed for typically attribute sets. If we roll our own bitset with a different backing type we can make it more efficient.